File: tareas/2/24190310/09. Es triángulo rectángulo/es_triangulo_rectangulo.py
import math

def distancia(x1, y1, x2, y2):
    dx=x2-x1
    dy=y2-y1
    return math.sqrt(dx*dx + dy*dy)

def es_triángulo_rectángulo(x1, y1, x2, y2, x3, y3):
    a=distancia(x1, y1, x2, y2)
    b=distancia(x1, y1, x3, y3)
    c=distancia(x2, y2, x3, y3)

    return a*a + b*b == c*c or a*a + c*c == b*b or c*c + b*b == a*a

def main():
  print(es_triángulo_rectángulo(0, 0, 0, 3, 4, 0))
  print(es_triángulo_rectángulo(0, 0, 0, 3, 4, 1))
if __name__ == "__main__":
   main()

File: ejercicios/09_10_11_12/es_triangulo_rectangulo/es_triangulo_rectangulo.cpp
#include <iostream>
#include <cmath>
using namespace std;

long double Distancia(long double x1, long double y1, long double x2, long double y2) {
    long double dx = x2 - x1;
    long double dy = y2 - y1;
    return sqrt(dx * dx + dy * dy);
}

bool EsTrianguloRectangulo(
    long double x1, long double y1,
    long double x2, long double y2,
    long double x3, long double y3
) {
    long double a = Distancia(x1, y1, x2, y2);
    long double b = Distancia(x1, y1, x3, y3);
    long double c = Distancia(x2, y2, x3, y3);

    return a * a + b * b == c * c ||
    a * a + c * c == b * b ||
    c * c + b * b == a * a;
}

int main(void) {
    cout << EsTrianguloRectangulo(0, 0, 0, 3, 4, 0) << endl;
    cout << EsTrianguloRectangulo(0, 0, 0, 3, 4, 1) << endl;
    return 0;
}